### PR TITLE
All the changes

### PR DESCRIPTION
**Pull request by a2**
Safe to merge now :grin:

### CODE REVIEW

**Comment by a2** [comment]
Also the tests still pass. :+1:

File: DANPromise/DANPromise.h [hunk 1]
```diff
@@ -6,34 +6,34 @@
...
 
-@property (nonatomic, strong) id __nullable result;
+@property (nonatomic, nullable, strong) id result;
 
 /**
- *  @return If the Promise has completed with a rejection.
+ *  If the Promise has completed with a rejection.
  */
-- (BOOL)isRejected;
+@property (nonatomic, getter=isRejected) BOOL rejected;
```
Comment (endocrimes) [inline DANPromise/DANPromise.h]: Should be readonly

File: DANPromise/DANPromise.h [hunk 2]
```diff
@@ -6,34 +6,34 @@
...
  */
-- (BOOL)isFulfilled;
+@property (nonatomic, getter=isFulfilled) BOOL fulfilled;
 
 /**
- *  @return If the promise was cancelled.
+ *  If the promise was cancelled.
  */
-- (BOOL)isCancelled;
+@property (nonatomic, getter=isCancelled) BOOL cancelled;
```
Comment (endocrimes) [inline DANPromise/DANPromise.h]: Should be readonly

File: DANPromise/DANPromise.h [hunk 3]
```diff
@@ -6,34 +6,34 @@
...
  */
-- (BOOL)isRejected;
+@property (nonatomic, getter=isRejected) BOOL rejected;
 
 /**
- *  @return If the Promise has completed successfully.
+ *  If the Promise has completed successfully.
  */
-- (BOOL)isFulfilled;
+@property (nonatomic, getter=isFulfilled) BOOL fulfilled;
```
Comment (endocrimes) [inline DANPromise/DANPromise.h]: Should be readonly

File: DANPromise/DANPromise_Private.h [hunk 4]
```diff
@@ -16,13 +18,16 @@ typedef NS_ENUM(NSInteger, DANPromiseState) {
 };
 
 @interface DANPromise ()
-@property (nonatomic, strong) dispatch_queue_t queue;
-@property (nonatomic, assign) DANPromiseState state;
-@property (nonatomic, strong) NSMutableArray *callbacks;
 
-@property (nonatomic, strong) dispatch_queue_t internalQueue;
+@property (nonatomic) dispatch_queue_t queue;
```
Comment (endocrimes) [inline DANPromise/DANPromise_Private.h]: storage type should be explicit

File: DANPromise/DANDeferredValue.m [hunk 5]
```diff
@@ -35,9 +31,9 @@ - (void)transitionToState:(DANPromiseState)state {
             
             shouldComplete = YES;
             
-            blocksToExecute = self.callbacks;
+            blocksToExecute = [self.callbacks copy];
             
-            self.callbacks = nil;
+            [self.callbacks removeAllObjects];
```
Comment (endocrimes) [inline DANPromise/DANDeferredValue.m:36]: It's technically safe to set this to nil after the state transition - nothing can touch it again.

File: DANPromise/DANDeferredValue.m [hunk 6]
```diff
@@ -13,16 +13,12 @@
 const NSInteger DANDeferredValueNilResultError = 1000;
 
 static NSError *NSErrorFromNilValue() {
+    NSDictionary *userInfo = @{NSLocalizedDescriptionKey: @"Unexpectedly found a nil value"};
```
Comment (endocrimes) [inline DANPromise/DANDeferredValue.m]: @{ NSLocalizedDescriptionKey: @"Unexpectedly found a nil value" };

File: DANPromise/DANPromise.m [hunk 7]
```diff
@@ -10,17 +10,13 @@
...
-
 @implementation DANPromise
 
 - (instancetype)init {
     self = [super init];
-    if (self) {
-        _callbacks = [NSMutableArray array];
-        _internalQueue = dispatch_queue_create("lt.danie.promises", 0);
-    }
+
```
Comment (endocrimes) [inline DANPromise/DANPromise.m]: Convention denotes that we still need an `if (self) { }` here, although is self is nil, something is seriously broken.

File: DANPromiseTests/DANTestCallback.h [hunk 8]
```diff
@@ -14,7 +15,7 @@
 @property (nonatomic, copy) DANPromiseSuccessBlock successBlock;
 @property (nonatomic, copy) DANPromiseErrorBlock errorBlock;
 
-@property (nonatomic, assign) NSInteger successBlockCallCount;
-@property (nonatomic, assign) NSInteger errorBlockCallCount;
+@property (nonatomic) NSInteger successBlockCallCount;
```
Comment (endocrimes) [inline DANPromiseTests/DANTestCallback.h]: explicit storage here as well please.

**Pull request merged**